externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/schema/RowSet.scala

```diff
@@ -165,6 +169,14 @@ object RowSet {
 
   private def toTColumn(rows: Seq[Row], ordinal: Int, logicalType: LogicalType): TColumn = {
     val nulls = new java.util.BitSet()
+    // for each column, determine the conversion class by sampling the first row
+    // if there's zero row, set empty as the column
+    var sampleField: Object = null
+    var i = -1
+    while (sampleField == null && i + 1 < rows.length) {
+      i += 1
+      sampleField = rows(i).getField(ordinal)
+    }
```

Comment:
Can we simply this code snippet via an `if/else` block?

Reply:
@yanghua I'd like to. Do you have a good idea? The problem is that we can't get the underlying conversion class unless we read one value.

Reply:
> @yanghua I'd like to. Do you have a good idea? I think the key is that we can't get the underlying conversion class unless we read one value.

I mean simplifying the way that we set the value for `sampleField ` variable.

The pseudo-code looks like this:

```
if (rows.length > 0) {
  sampleField = rows(0)..getField(ordinal)
}
```

Can we remove the `while` block and `i` counter?

Reply:
But if we get a null value, we still can't determine its conversion class. We need a non-null value here, that's what the loop is for.

Reply:
> But if we get a null value, we still can't determine its conversion class. We need a non-null value here, that's what the loop is for.

Oh... I misunderstood, based on your comment. Now, I suggest, you change your comment:

`by sampling the first row` -> `by sampling the first non-null row`.

So the suggestion may be to simplify the code style about how to find the first non-null value right?

May you reference this [answer](https://stackoverflow.com/a/8715730) in the StackOverflow?

Reply:
Sorry for the misleading outdated comment. Updating the comment and the codes.